aware-core/src/main/java/com/aware/Bluetooth.java

```diff
@@ -303,33 +349,6 @@ private void save_bluetooth_device(BluetoothAdapter btAdapter) {
         if (sensorBT != null && ! sensorBT.isClosed()) sensorBT.close();
     }
 
-    public static class BackgroundService extends IntentService {
-        public BackgroundService() {
-            super(TAG + " background service");
-        }
-
-        @Override
-        protected void onHandleIntent(Intent intent) {
-            if (intent.getAction().equals(ACTION_AWARE_BLUETOOTH_REQUEST_SCAN)) {
-                if (!bluetoothAdapter.isDiscovering()) {
-                    if (bluetoothAdapter.isEnabled()) {
-                        bluetoothAdapter.startDiscovery();
-                    } else {
-                        //Bluetooth is off
-                        if (Aware.DEBUG) Log.d(TAG, "Bluetooth is turned off...");
-                        ContentValues rowData = new ContentValues();
-                        rowData.put(Bluetooth_Data.DEVICE_ID, Aware.getSetting(getApplicationContext(), Aware_Preferences.DEVICE_ID));
-                        rowData.put(Bluetooth_Data.TIMESTAMP, System.currentTimeMillis());
-                        rowData.put(Bluetooth_Data.BT_NAME, "disabled");
-                        rowData.put(Bluetooth_Data.BT_ADDRESS, "disabled");
-                        rowData.put(Bluetooth_Data.BT_LABEL, "disabled");
-                        getContentResolver().insert(Bluetooth_Data.CONTENT_URI, rowData);
-                    }
-                }
-            }
-        }
-    }
-
     private static void notifyMissingBluetooth(Context c) {
```

Comment:
This is related to the issue of showing a notification about the BT when it is enabled. Maybe extend this push to also fix the condition in which this function is called?